datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -686,6 +686,37 @@ public Datastream get(String name) {
     return stream;
   }
 
+  /**
+   * Finds the task's assignment node which hosts the given task 
+   * @param pathKeys Datastream resource key
+   * @param task Datastream task name
+   */
+  @Action(name = "getTaskAssignment")
+  public String getTaskAssignment(@PathKeysParam PathKeys pathKeys, @ActionParam("task") String task) {
+    // Get datastream
+    String datastreamName = pathKeys.getAsString(KEY_NAME);
+
+    Datastream stream;
+    String assignedInstance = null;
+    try {
+      LOG.info("Get assigned instance for task: {}", task);
+      _dynamicMetricsManager.createOrUpdateMeter(CLASS_NAME, GET_CALL, 1);
+      stream = _store.getDatastream(datastreamName);
+      assignedInstance = _store.getRawData(_store.getConnectorTaskPath(stream.getConnectorName(), task));
+    } catch (Exception e) {
+      _dynamicMetricsManager.createOrUpdateMeter(CLASS_NAME, CALL_ERROR, 1);
+      _errorLogger.logAndThrowRestLiServiceException(HttpStatus.S_500_INTERNAL_SERVER_ERROR,
+          "Failed to get assigned instance hosting task: " + task, e);
+    }
+
+    if (assignedInstance == null) {
```

Comment:
what if the assignedInstance is empty for some reason.

Reply:
I handled this case in the tool logic which calls this endpoint, but thinking about it again, it makes more sense to handle that here.